src/payment/strategies/paypal-commerce/paypal-commerce-payment-strategy.ts

```diff
@@ -93,12 +111,29 @@ export default class PaypalCommercePaymentStrategy implements PaymentStrategy {
     }
 
     async deinitialize(): Promise<InternalCheckoutSelectors> {
+        clearInterval(this._pollingInterval);
```

Comment:
you have `clearInterval(this._pollingInterval);` in quite a few places.

Maybe you can have a private method `_deinitiazePollingMechanism()` ?
```ts
private _deinitiazePollingMechanism() {
  if (this._pollingInterval) {
    clearInterval(this._pollingInterval);
    this._pollingInterval = null;
  }
}
```

Reply:
fixed